### PR TITLE
feat(lsp): get_border_size(): support repeating border char list

### PR DESCRIPTION
**Pull request by zeertzjq**
Support any border char list length that is a divisor of 8.

### CODE REVIEW

**Comment by zeertzjq** [comment]
Maybe this function can become a public API to save plugin authors some time?

**Comment by zeertzjq** [comment]
Also I think `strdisplaywidth` doesn't make much sense since characters occupying multiple cells are not allowed in border and Neovim doesn't support zero-width characters.

File: runtime/lua/vim/lsp/util.lua [hunk 1]
```diff
@@ -57,7 +57,12 @@ local function get_border_size(opts)
     end
     height, width = unpack(border_size[border])
   else
+    if 8 % #border ~= 0 then
+      error("floating preview border is not correct. Please refer to the docs |vim.api.nvim_open_win()|"
```
Comment (justinmk) [inline runtime/lua/vim/lsp/util.lua:61]:
```suggestion
      error("floating preview border is not correct. Refer to the docs |vim.api.nvim_open_win()|"
```

please is never needed in messages, it's noise

Comment (clason) [reply runtime/lua/vim/lsp/util.lua:61]: But a "see" in front of the ":help" would read better.

Comment (justinmk) [reply runtime/lua/vim/lsp/util.lua:61]: not really, that is not the convention in messages. The messages are seen many times and don't need to explain the same thing over and over. The convention is learned only once.

**Review by justinmk — APPROVED** [review]
would prefer adding a test. it doesn't need to be elaborate, just call the function somehow and check some property. also trigger the error condition.

**Pull request merged**